specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2020-06-01-preview/nova.json

```diff
@@ -0,0 +1,1051 @@
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "body",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ClusterResource"
+            },
+            "description": "Parameters to provide for specifying the managed Cassandra cluster."
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "The managed Cassandra cluster is being updated. Poll for provisioningState=Succeeded to find when the update completes.",
+            "schema": {
+              "$ref": "#/definitions/ClusterResource"
+            }
+          },
+          "200": {
+            "description": "The managed Cassandra cluster has been updated.",
+            "schema": {
+              "$ref": "#/definitions/ClusterResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DefaultErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/cassandraClusters/{clusterName}/repair": {
+      "post": {
+        "operationId": "CassandraCluster_RequestRepair",
+        "description": "Request that repair begin on this cluster as soon as possible.",
+        "x-ms-examples": {
+          "CosmosDBManagedCassandraRepair": {
+            "$ref": "./examples/CosmosDBManagedCassandraRepair.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/clusterNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "The repair request was successfully received."
+          },
+          "409": {
+            "description": "Repair is disabled for this cluster, so no repair request can be accepted."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DefaultErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/cassandraClusters/{clusterName}/instanceView": {
```

Comment:
>instanceView [](start = 135, length = 12)

this is the url pattern for a list call - if a cluster has only 1 instanceview you must model it as a /instanceViews/default api - and also support /instanceViews which will return the collection format with 1 resource

Reply:
if the response schema does not represent a resource (id, name, type, properties) - model it as a post instead with name fetchInstanceView

---
In reply to: [573443410](https://github.com/Azure/azure-rest-api-specs/pull/12861#discussion_r573443410) [](ancestors = 573443410)

Reply:
Switch it to a POST of fetchInstanceView.

Reply:
Actually, we're going with the GET pattern, so switched it to the new path.